code/controllers/subsystem/ipintel.dm

```diff
@@ -3,13 +3,374 @@ SUBSYSTEM_DEF(ipintel)
 	wait = 1
 	flags = SS_NO_FIRE
 	init_order = INIT_ORDER_XKEYSCORE // 10
-	var/enabled = 0 //disable at round start to avoid checking reconnects
+	// Are we enabled? Auto disable at world init to avoid checking reconnects
+	var/enabled = FALSE
 	var/throttle = 0
 	var/errors = 0
 
 	var/list/cache = list()
 
-/datum/controller/subsystem/ipintel/Initialize(timeofday, zlevel)
-	enabled = 1
-	. = ..()
+/datum/controller/subsystem/ipintel/Initialize(timeofday)
+	enabled = TRUE
+	return ..()
 
+
+// Represents an IP intel holder datum
+/datum/ipintel
+	/// The IP being checked
+	var/ip
+	/// The current rating, 0-1 float.
+	var/intel = 0
+	/// Whether this was loaded from the cache or not
+	var/cache = FALSE
+	/// How many minutes ago it was cached
+	var/cacheminutesago = 0
+	/// The date it was cached
+	var/cachedate = ""
+	/// The real time it was cached
+	var/cacherealtime = 0
+
+/datum/ipintel/New()
+	cachedate = SQLtime()
+	cacherealtime = world.realtime
+
+/datum/ipintel/proc/is_valid()
+	. = FALSE
+	if(intel < 0)
+		return
+	if(intel <= GLOB.configuration.ipintel.bad_rating)
+		if(world.realtime < cacherealtime + (GLOB.configuration.ipintel.hours_save_good HOURS))
+			return TRUE
+	else
+		if(world.realtime < cacherealtime + (GLOB.configuration.ipintel.hours_save_bad HOURS))
+			return TRUE
+
+
+
+/**
```

Comment:
I can't comment on the rest of this since file since I have no idea what any of it does, but are the three spaces between each of these procs intentional? The standard usually seems to be 1-2.

Reply:
I wanted more separation, especially when filling in my documentation